### PR TITLE
Add target 'make help'

### LINKED ISSUE #1318: Add "make help"
Our Makefiles should have a target ``help`` that shows short documentation about supported targets and variables.

### PR DESCRIPTION
**Pull request by cmarqu**
Closes #1318.

### CODE REVIEW

**Comment by ktbarrett** [comment]
There is a `make help` in the root level `Makefile`. Should that be updated?

**Comment by themperek** [comment]
@ktbarrett I would rather be in favour to move top level makefile to tests folder.

**Comment by cmarqu** [comment]
> There is a `make help` in the root level `Makefile`. Should that be updated?

I hadn't noticed that, but it is a target that only helps for working with/testing cocotb itself (but indeed can use an update).
The help target of this PR is shown when you are developing your own testbench.

File: cocotb/share/makefiles/Makefile.sim [hunk 1]
```diff
@@ -30,8 +30,71 @@
 # This is a simple wrapper Makefile to pull in the appropriate Makefile for
 # the desired simulator and set up common paths for the sims to use to build with
 
+.PHONY: all
 all: sim
 
+# NOTE: make sure to keep "helpmsg" aligned with documentation/source/building.rst
+# Also keep it at 80 chars.
```
Comment (eric-wieser) [inline cocotb/share/makefiles/Makefile.sim]: Keeping this aligned sounds like a lot of effort - can we just bundle this RST file with the install, and read it in here?

Comment (cmarqu) [reply cocotb/share/makefiles/Makefile.sim]: "Aligned" does not mean "duplicate", the text in ``helpmsg`` is shortened to be the essence of the description for people who want to refresh their memory and may leave out details.
I do not expect this to change often anymore, after all, we want to get rid of Makefiles at some point. Once we have them replaced with Python scripts, we can use ``sphinx-argparse`` to get the argument descriptions into the documentation.

**Comment by cmarqu** [comment]
> There is a `make help` in the root level `Makefile`. Should that be updated?

This has been done in the meantime.

File: cocotb/share/makefiles/Makefile.sim [hunk 2]
```diff
@@ -30,8 +30,71 @@
 # This is a simple wrapper Makefile to pull in the appropriate Makefile for
 # the desired simulator and set up common paths for the sims to use to build with
 
+.PHONY: all
```
Comment (imphil) [inline cocotb/share/makefiles/Makefile.sim]: That's not really related to this PR, it should be a separate commit at least.

File: cocotb/share/makefiles/Makefile.sim [hunk 3]
```diff
@@ -30,8 +30,71 @@
...
+COCOTB_ATTACH             Pause time value in seconds before the simulator start
+COCOTB_ENABLE_PROFILING   Performance analysis of the Python portion of cocotb
+COCOTB_HOOKS              Comma-separated module list to be executed before test
+COCOTB_LOG_LEVEL          Default logging level (default INFO)
+COCOTB_RESOLVE_X          How to resolve X, Z, U, W on integer conversion
+COCOTB_SCHEDULER_DEBUG    Enable additional output of coroutine scheduler
+COVERAGE                  Report Python coverage (also HDL for some simulators)
+MEMCHECK                  HTTP port to use for debugging Python memory usage
+
+    For details, see https://cocotb.readthedocs.io/en/latest/building.html
```
Comment (imphil) [inline cocotb/share/makefiles/Makefile.sim]: If you want to be fully correct use a versioned link here (e.g. https://cocotb.readthedocs.io/en/v1.3.0/). But that might be too much work for little gain.

File: cocotb/share/makefiles/Makefile.sim (hunk 2 above)
Comment (cmarqu) [reply cocotb/share/makefiles/Makefile.sim]: Done.

File: cocotb/share/makefiles/Makefile.sim (hunk 3 above)
Comment (cmarqu) [reply cocotb/share/makefiles/Makefile.sim]: I may do this in a future PR. If the output of ``cocotb-config -v`` contains "dev", point to "latest", otherwise derive from the returned version.

**Comment by cmarqu** [comment]
> If you really want command-line accessible documentation, we can think about building man pages from our docs using Sphinx.

I like that idea and will open a new issue for it. It doesn't solve my "problem" of wanting immediately useful reminders of what a certain variable was called again though.
